pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -432,7 +435,7 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 		}
 
 		_, err = attributestags.ReplaceAll(is.networkClient, "ports", port.ID, attributestags.ReplaceAllOpts{
-			Tags: clusterTags}).Extract()
```

Comment:
this if for network client? why we tag a server tag to a network ? for example ,one server tag as 't1' we have to tag all ports with 't1'?? IMHO we should not have those kind of logic unless we have very specified reason to do so

if user need to tag something on their own, they can tag in their 
- network
section...

Reply:
I actually think it's fine to propagate the server tags down to the network as it creates a relationship between the resources, it makes it easier to filter, and to clean up later on. To be clear, I think this is a good thing only within the context of the cluster-api-provider-openstack.

That said, I didn't want to touch this part of the code much as it's not really the main goal of this PR

Reply:
I agree, I think we should use one PR for one thing, this add can be another PR and more opionion got